File: src/main/java/net/runelite/client/plugins/oneclickagility/ObstacleArea.java
package net.runelite.client.plugins.oneclickagility;

import lombok.Getter;
import lombok.Setter;
import dev.hoot.api.game.Game;
import meteor.Main;
import net.runelite.api.GameObject;
import net.runelite.api.Locatable;
import net.runelite.api.MenuAction;
import net.runelite.api.MenuEntry;
import net.runelite.api.TileObject;
import net.runelite.api.coords.WorldPoint;

public class ObstacleArea
{
    private final int minX;
    private final int maxX;
    private final int minY;
    private final int maxY;
    private final int z;
    @Getter
    private final int nextObstacleID;
    @Setter
    private TileObject nextObstacle;

    ObstacleArea(int minX,int maxX, int minY, int maxY,int z, int nextObstacleID)
    {
        this.minX = minX;
        this.maxX = maxX;
        this.minY = minY;
        this.maxY = maxY;
        this.z = z;
        this.nextObstacleID = nextObstacleID;
    }

    public boolean containsObject(Locatable locatable)
    {
        return containsObject(locatable.getWorldLocation());
    }

    public boolean containsObject(WorldPoint worldPoint)
    {
        return containsObject(worldPoint.getX(),worldPoint.getY(), worldPoint.getPlane());
    }

    private boolean containsObject(int x, int y, int z)
    {
        return x >= minX && x <= maxX && y >= minY && y <= maxY && z == this.z;
    }

    public MenuEntry createMenuEntry()
    {
        if (nextObstacle != null)
        {
            if (nextObstacle.distanceTo(Game.getClient().getLocalPlayer().getWorldLocation()) < 1)
                return null;
            return Main.INSTANCE.getClient().createMenuEntry(nextObstacle.getRawActions()[0], nextObstacle.getName(),nextObstacle.getId(), MenuAction.GAME_OBJECT_FIRST_OPTION.getId(), getObjectParam(nextObstacle),getObjectParam1(nextObstacle),true);
        }
        return null;
    }

    private int getObjectParam(Locatable gameObject)
    {
        if (gameObject instanceof GameObject)
        {
            return ((GameObject) gameObject).getSceneMinLocation().getX();
        }
        return(gameObject.getLocalLocation().getSceneX());
    }

    private int getObjectParam1(Locatable gameObject)
    {
        if (gameObject instanceof GameObject)
        {
            return ((GameObject) gameObject).getSceneMinLocation().getY();
        }
        return(gameObject.getLocalLocation().getSceneY());
    }
}

File: client/src/main/kotlin/meteor/plugins/oneclickagility/ObstacleArea.kt
package meteor.plugins.oneclickagility

import dev.hoot.api.game.Game
import meteor.Main.client
import net.runelite.api.*
import net.runelite.api.coords.WorldPoint

class ObstacleArea internal constructor(
    private val minX: Int,
    private val maxX: Int,
    private val minY: Int,
    private val maxY: Int,
    private val z: Int,
    val nextObstacleID: Int
) {
   var nextObstacle: TileObject? = null
    fun containsObject(locatable: Locatable): Boolean {
        return containsObject(locatable.worldLocation)
    }

    fun containsObject(worldPoint: WorldPoint): Boolean {
        return containsObject(worldPoint.x, worldPoint.y, worldPoint.plane)
    }

    private fun containsObject(x: Int, y: Int, z: Int): Boolean {
        return x in minX..maxX && y >= minY && y <= maxY && z == this.z
    }

    fun createMenuEntry(): MenuEntry? {
        return if (nextObstacle != null) {
            if (nextObstacle!!.distanceTo(Game.getClient().localPlayer!!.worldLocation) < 1) null else client.createMenuEntry(
                nextObstacle!!.rawActions!![0],
                nextObstacle!!.name,
                nextObstacle!!.id,
                MenuAction.GAME_OBJECT_FIRST_OPTION.id,
                getObjectParam(nextObstacle!!),
                getObjectParam1(nextObstacle!!),
                true
            )
        } else null
    }

    private fun getObjectParam(gameObject: Locatable): Int {
        return if (gameObject is GameObject) {
            gameObject.sceneMinLocation.x
        } else gameObject.localLocation.sceneX
    }

    private fun getObjectParam1(gameObject: Locatable): Int {
        return if (gameObject is GameObject) {
            gameObject.sceneMinLocation.y
        } else gameObject.localLocation.sceneY
    }
}